tests/integration/scale_test.go

```diff
@@ -37,5 +33,33 @@ func scaleTests(t *testing.T, when spec.G, it spec.S) {
 			assert.Equal(t, 0, service.GetAvailableReplicas())
 			assert.Equal(t, 0, service.GetScale())
 		})
+
+		it.After(func() {
+			service.Remove()
+		})
+
 	}, spec.Parallel())
+
+	when.Focus("services should scale up from 0", func() {
+		var service testutil.TestService
+		it.Before(func() {
+			service.Create(t, "--scale", "0-3", "sangeetha/mytestcontainer")
+
+		})
+		it("service should never return 503 'no healthy upstream'", func() {
+			// check that service can scale up
+			service.GetEndpointResponse()
+			for service.GetScale() > 0 {
+				time.Sleep(1 * time.Second)
```

Comment:
@rancher-max Do we have any functions in testutil that can wait till a service naturally scales to zero? Going to add one I think

Reply:
`service.WaitForScaleDown` should do it for you! You also might want to double check the test in tests/validation/autoscale_test.go that does autoscale down to 0 to make sure you're not duplicating.

I removed it from integration because it was a bit flakey on the "scale back up" part and took more time due to the amount of tests running in parallel